core/dbt/contracts/graph/model_config.py

```diff
@@ -355,10 +356,38 @@ class SourceConfig(BaseConfig):
 
 
 @dataclass
-class NodeConfig(BaseConfig):
+class NodeAndTestConfig(BaseConfig):
```

Comment:
This is a superclass of `NodeConfig` and `TestConfig`. I don't see it used explicitly beyond the inheritance, so is this just for defining common fields in one place? If so, we might want to make this an `abstract base class` so it's clear there will never be any instances of this type floating around.

Reply:
I'm pretty sure that it can't be an abstract base class because it's an intermediate class in the hierarchy, not an actual base class.

Reply:
ugh that's good point. nevermind then.